libs/data-mapper/src/lib/components/inputDropdown/InputDropdown.tsx

```diff
@@ -289,10 +296,15 @@ export const InputDropdown = (props: InputDropdownProps) => {
 
       const inputs = connectionDictionary[key].inputs[0];
       const sourceNode = inputs && inputs[0];
-      const nodeName =
-        node.key === indexPseudoFunctionKey && isConnectionUnit(sourceNode) && isSchemaNodeExtended(sourceNode.node)
-          ? calculateIndexValue(sourceNode.node)
-          : getFunctionOutputValue(fnInputValues, node.functionName);
+      let nodeName: string;
+      if (node.key === indexPseudoFunctionKey && isConnectionUnit(sourceNode) && isSchemaNodeExtended(sourceNode.node)) {
```

Comment:
*Line 302* Does this need the isConnectionUnit/isSchemaNodeExtended check too?

Reply:
Nope. Those checks are type converting the sourceNode and we don't need to do that in 302